### PR TITLE
[SPARK-45588][PROTOBUF][CONNECT][MINOR] Scaladoc improvement for StreamingForeachBatchHelper

### PR DESCRIPTION
**Pull request by rangadi**
### What changes were proposed in this pull request?

Couple of minor improvements to `StreamingForeachBatchHelper`:

  * Make `RunnerCleaner` private and add ScalaDoc.
  * Update contract for `pythonForeachBatchWrapper()` to inform that call should eventually should `close()` the `AutoClosable` returned. 
  
In addition, it also fixes a flake in Protobuf unit test.

### Why are the changes needed?
  - Code readability improvement.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
 - Existing tests.
 - For protobuf suite, verified with seed set to '399'. It fails before this PR and passes after.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by rangadi** [comment]
cc: @WweiL

**Comment by srowen** [comment]
Looks OK, just get tests to pass

File: connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufCatalystDataConversionSuite.scala [hunk 1]
```diff
@@ -137,7 +137,7 @@ class ProtobufCatalystDataConversionSuite
       while (
         data != null &&
         (data.get(0) == defaultValue ||
-          (dt == BinaryType &&
+          (dt.fields(0).dataType == BinaryType &&
```
Comment (rangadi) [inline connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufCatalystDataConversionSuite.scala:140]: This fixes flake reported in https://github.com/apache/spark/pull/43413#issuecomment-1769286752
The type check was incorrect. So when the random value was empty array, we didn't skip it. Original intention is to skip default values for types. Verified with manually setting the seed to 399 (at line 128).
cc: @HeartSaVioR 

image

Comment (HeartSaVioR) [reply connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufCatalystDataConversionSuite.scala:140]: Ah OK that's actually failing consistently for specific value but the test itself is using random seed hence flaky...

**Review by HeartSaVioR — APPROVED** [review]
+1 pending CI. Please mention in PR description that you've run the test locally with seed 399.

**Comment by rangadi** [comment]
@HeartSaVioR all the tests passes. Updated "testing" section in the description.

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**